spec/requests/users_controller_spec.rb

```diff
@@ -5273,6 +5273,84 @@ def users_found
     end
   end
 
+  describe "#lookup" do
+    fab!(:user_1) { Fabricate(:user) }
+    fab!(:user_2) { Fabricate(:user) }
+    fab!(:user_3) { Fabricate(:user) }
+
+    before { sign_in(user) }
+
+    it "lookups users by username" do
+      get "/u/lookup/users.json",
+          params: {
+            usernames: [user_1.username, user_2.username, user_3.username],
+          }
+
+      expect(response.status).to eq(200)
+      users = response.parsed_body["users"]
+      expect(users.length).to be(3)
+
+      ids = users.map { |user| user["id"] }
+      expect(ids).to include(user_1.id)
+      expect(ids).to include(user_2.id)
+      expect(ids).to include(user_3.id)
```

Comment:
Maybe use `contrains_exactly` instead of matching ids one by one?

Reply:
Fixed in 145c787f96d748030ce0cba9e3301db1c9b91be7.